Xamarin.Forms.Platform.iOS/Material/MaterialProgressBarRenderer.cs

```diff
@@ -1,87 +1,185 @@
+
+			var cs = MaterialColors.Light.CreateColorScheme();
+			return new BasicColorScheme(
+				cs.PrimaryColor,
+				cs.PrimaryColor.ColorWithAlpha(BackgroundAlpha),
+				cs.PrimaryColor);
+		}
+
+		protected virtual void ApplyTheme()
+		{
+			// TODO: Fix this once Google implements the new way.
+
+#pragma warning disable CS0618 // Type or member is obsolete
+			ProgressViewColorThemer.ApplyColorScheme(_colorScheme, Control);
+#pragma warning restore CS0618 // Type or member is obsolete
 		}
 
+		protected override MProgressView CreateNativeControl()
+		{
+			return new MProgressView();
+		}
 
-		public override SizeF SizeThatFits(SizeF size)
+		public override CGSize SizeThatFits(CGSize size)
 		{
 			var result = base.SizeThatFits(size);
 			var height = result.Height;
 
-			if(height == 0)
+			if (height == 0)
 			{
-				if(System.nfloat.IsInfinity(size.Height))
-				{
-					height = 5;
-				}
+				if (nfloat.IsInfinity(size.Height))
+					height = 4;
 				else
-				{
 					height = size.Height;
-				}
-
 			}
-			return new SizeF(10, height);
+
+			return new CGSize(10, height);
 		}
 
 		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
 		{
 			base.OnElementPropertyChanged(sender, e);
 
+			var updatedTheme = false;
 			if (e.PropertyName == ProgressBar.ProgressColorProperty.PropertyName)
+			{
 				UpdateProgressColor();
+				updatedTheme = true;
+			}
 			else if (e.PropertyName == ProgressBar.ProgressProperty.PropertyName)
+			{
 				UpdateProgress();
+			}
+
+			if (updatedTheme)
+				ApplyTheme();
 		}
 
 		protected override void SetBackgroundColor(Color color)
 		{
-			base.SetBackgroundColor(color);
+			// Do not call base to avoid the actual background view changing color.
+			//base.SetBackgroundColor(color);
 
-			if (Control == null || color == Color.Default)
+			if (Control == null)
 				return;
 
-			Control.TrackTintColor = color.ToUIColor();
+			UpdateAllColors();
+			ApplyTheme();
 		}
 
 		void UpdateProgressColor()
 		{
-			if (Element.ProgressColor == Color.Default)
-				return;
+			UpdateAllColors();
+		}
+
+		void UpdateAllColors()
+		{
+			// TODO: Fix this once Google implements the new way.
+			//       Right now, copy what is done with the activity indicator.
 
-			Control.ProgressTintColor = Element.ProgressColor.ToUIColor();
+			Color progressColor = Element.ProgressColor;
+			Color backgroundColor = Element.BackgroundColor;
+
+			if (progressColor.IsDefault && backgroundColor.IsDefault)
+			{
+				// reset everything to defaults
+				_colorScheme = new BasicColorScheme(
+					_defaultColorScheme.PrimaryColor,
+					_defaultColorScheme.PrimaryLightColor,
+					_defaultColorScheme.PrimaryColor);
+			}
+			else if (progressColor.IsDefault && !backgroundColor.IsDefault)
+			{
+				// handle the case where only the background is set
+				var background = backgroundColor.ToUIColor();
+
+				// TODO: Potentially override background alpha to match material design.
```

Comment:
Same question as above - do we need to address this TODO?

Reply:
Not for this PR we don't 
Once we get this merged we can start working on the PRs for these todos

Reply:
Will fixing these result in behavior changes? That is, is we leave them as-is and don't get around to addressing them before this gets a public release, are we going to be stuck supporting a legacy behavior?

Reply:
> That is, is we leave them as-is and don't get around to addressing them before this gets a public release, are we going to be stuck supporting a legacy behavior?

I would say there's not really an option to not get around to the TODOs before calling this a public release. The material components are still experimental so they all currently have "//TODO's" like this (known and unknown) that need to be addressed before public release.  

This PR has a lot of general infrastructure that apply across all the controls.  It also introduces a bunch of new ones that we can start getting feedback on.The rest of the PRs will be more control by control addressing these "TODOs" and making them all releasable